src/core/aws-utils.adb

```diff
@@ -883,34 +887,64 @@ package body AWS.Utils is
    -- Random --
    ------------
 
-   function Random return Random_Integer is
+   function Random (Task_Safe : Boolean := False) return Random_Integer is
    begin
-      return Integer_Random.Random (Random_Generator);
+      return (if Task_Safe
+              then Random_Protecter.Random
+              else Integer_Random.Random (Random_Generator));
    end Random;
 
+   ----------------------
+   -- Random_Protecter --
+   ----------------------
+
+   protected body Random_Protecter is
+
+      ------------
+      -- Random --
+      ------------
+
+      function Random return Random_Integer is
+      begin
+         return Integer_Random.Random (Random_Generator);
+      end Random;
+
+   end Random_Protecter;
+
+   ------------------
+   -- Random_Reset --
+   ------------------
+
+   procedure Random_Reset (Initiator : Integer) is
```

Comment:
Usually this is called a Seed for random numbers. (Initiator -> Seed).

Reply:
Done